tests/01-known-values.sh

```diff
@@ -3,25 +3,22 @@
 ### Constants
 # The scenario command requires a lot of memory, so valgrind is only enabled
 # if $USE_VALGRIND > 1.
-scenario_valgrind_min=2
+c_valgrind_min=2
 
 ### Actual command
 scenario_cmd() {
-	$val_cmd $bindir/tests/test_scrypt \
-		1> $out/$basename-stdout.txt
-	cmd_retval=$?
-
-	echo "$cmd_retval"
-}
-
-### Check output
-scenario_check() {
-	retval=$1
+	# Run the binary which tests known input/output strings.
+	setup_check_variables
+	(							\
+		$c_valgrind_cmd $bindir/tests/test_scrypt	\
+			1> $s_basename-stdout.txt		\
+	; echo $? > $c_exitfile )
```

Comment:
Err, if you click on `[...]`, you'll see the commit message.

"""
Note that the awkward

        ( \
        	cmd... \
        ; echo $? > $c_exitfile )

construct is used because ${cmd} might not be exiting before we want the test suite to advance to the next command.  This likely won't apply to test scenarios in scrypt, but when we apply this code to spipe, being able to grab the return code of a server which began 10 shell commands ago is absolutely necessary.  I thought the test suite (as a whole) would be easier to understand if I used that construct everywhere, instead of only for certain scenarios.
"""

I added the extra tab in order to distinguish the "real command" from the boilerplate, but I'm content to remove that if you prefer it removed.